students/Deniss_Semcovs/Lesson05/mailroom_part03.py

```diff
@@ -0,0 +1,153 @@
+            break
+        else:
+            for i in data:
+                if full_name == i[0]:
+                    i.append(int(input("Please enter the donation amount: ")))            
+                    print("Donation has been submitted! You can see new donation total in the report.")
+                    break          
+            else:
+                try:
+                    new_donation = int(input("This is a new donor, please enter the donation amount: "))
+                    if new_donation is None or False:
+                        pass
+                    else:
+                        data.append([full_name, new_donation])
+                        print("Letter for the donor: ")
+                        print("Hello {}, thank you for your generous donation of ${} to support our cause.".format(full_name, new_donation))
+                    break
+                except (ValueError):
+                    valueerror()
+                except KeyboardInterrupt:
+                    keyboardinterrupt()
+
+def create_report():
+    tags = ['Donor Name', 'Total Given', 'Num Gifts', 'Average Gift']
+    new_data = dict_donor()
+    print("{:20s}| {:10s} | {:10s} | {:10s}".format(*tags))
+    print(("-")*61)
+    for i in new_data:
+        data2 = (i["name"], i["don"], i["count"], i["average"])
+        print("{:20s} $ {:10d}   {:10d}  $ {:.2f}".format(*data2))
+
+def send_ty_note_all():
+    new_data = dict_donor()
+    for i in new_data:
+        letter_file = os.path.join(tempfile.gettempdir(), (i["name"]+".txt"))
+        with open(letter_file, "w") as f:
+            f.write("\nDear "+i["name"]+",")
+            f.write("\n"*2)
+            f.write(" "*8+"Thank you for your very kind donation of $"+str(i["don"])+".")
+            f.write("\n"*2)
+            f.write(" "*8+"It will be put to very good use.")
+            f.write("\n"*2)
+            f.write(30*" "+"Sincerely,")
+            f.write(34*" "+"-The Team")
+    print("<<< Letters have been created in the temp folder! >>>")
+
+def quit():
+    print("Quiting this menu now!")
+    return "exit menu"
+
+# After changing data source from list to dict 
+# function does not scan through all names while "else" option exist
+def send_ty_note():
+    try:
+        new_data = dict_donor()
+        don_name = input("Enter donor's name: ")
+        for i in new_data:
+            if don_name == i["name"]:
+                print("Hello {}, thank you for your generous donation of ${} to support our cause.".format(i['name'], i['don']))
+            # I had to remove "else:" because function stoped checking values 
+            # after first dictionary, please let know what did i do wrong here?
+            """
+                break
+            else:
+                print("The donor's name is not on the list! Please add a new donor through 'Add new donation' option.")
+                print(i["name"])
+                break
+                """
```

Comment:
this is because you have a `break` here - to find donor not in your list you want your `else` to be on the same indent level as `for`